hermes_spani/calibration/calibration.py

```diff
@@ -2,53 +2,205 @@
+
+    # check if level 0 binary file, if so call appropriate functions
+    if (
+        file_metadata["instrument"] == hermes_spani.INST_NAME
+        and file_metadata["level"] == "l0"
+    ):
+        #  data = parse_l0_sci_packets(data_filename)
+        data = {}
+        # test opening the file
+        with open(data_filename, "r") as fp:
+            pass
+        level1_filename = l0_sci_data_to_cdf(data, data_filename)
+        output_filename = level1_filename
+    elif (
+        file_metadata["instrument"] == hermes_spani.INST_NAME
+        and file_metadata["level"] == "l1"
+    ):
+        # generate the quicklook data
+        #
+        # the following shows an example flow for calibrating a file
+        # data = read_file(data_filename)
+        # calib_file = get_calibration_file(data_filename)
+        # if calib_file is None:
+        #    raise ValueError(f"Calibration file for {data_filename} not found.")
+        # else:
+        #    calib_data = read_calibration_file(calib_file)
+
+        # test opening the file
+        with open(data_filename, "r") as fp:
+            pass
+
+        # now that you have your calibration data, you can calibrate the science data
+        ql_filename = data_filename.parent / create_science_filename(
+            file_metadata["instrument"],
+            file_metadata["time"],
+            "ql",
+            file_metadata["version"],
         )
-    )
 
-    calib_file = get_calibration_file(data_filename)
-    if calib_file is None:
-        raise ValueError("Calibration file for {} not found.".format(data_filename))
+        # write your cdf file below
+        # create an empty file for testing purposes
+        with open(data_filename.parent / ql_filename, "w"):
+            pass
+
+        # example log messages
+        log.info(f"Despiking removing {random.randint(0, 10)} spikes")
+        log.warning(f"Despiking could not remove {random.randint(1, 5)}")
+        output_filename = ql_filename
     else:
-        calib_data = read_calibration_file(calib_file)
+        raise ValueError(f"The file {data_filename} is not recognized.")
 
-    return None
+    return output_filename
+
+
+def parse_l0_sci_packets(data_filename: Path) -> dict:
+    """
+    Parse a level 0 spani binary file containing CCSDS packets.
+
+    Parameters
+    ----------
+    data_filename: str
+        Fully specificied filename
 
+    Returns
+    -------
+    result: dict
+        A dictionary of arrays which includes the ccsds header fields
 
-def get_calibration_file(data_filename, time=None):
+    Examples
+    --------
+    >>> import hermes_spani.calibration as calib
+    >>> data_filename = "hermes_MAG_l0_2022339-000000_v0.bin"
+    >>> data = calib.parse_spani_sci_packets(data_filename)  # doctest: +SKIP
+    """
+    log.info(f"Parsing packets from file:{data_filename}.")
+
+    pkt = ccsdspy.FixedLength.from_file(
+        os.path.join(hermes_spani._data_directory, "MAG_sci_packet_def.csv")
```

Comment:
Why is MAG_sci_packet_def.csv defined as a constant?  Shouldn't this come fromt he variable data_filename?

Reply:
Oh, sorry. Copy and paste from NEMISIS that I forgot to remove here.